src/javascript/app/ng-wig/ngWigEditableDirective.js

```diff
@@ -14,7 +14,25 @@ angular.module('ngWig')
         ngModelController.$setViewValue($element.html());
       }
 
-      $element.bind('blur keyup change paste focus click', function() {
+      var eventsToBind = [
+        'blur',
+        'keyup',
+        'change',
+        'focus',
+        'click'
+      ];
+
+      if (angular.isFunction(scope.onPaste)) {
+        $element.on('paste', function(e) {
+          scope.onPaste(e, $element.html()).then(function(val) {
```

Comment:
Yea I am sure. Usually this usecase should not require to be async but in
case someone needs it for $timeout for example it will work as well.

Stepan Suvorov <notifications@github.com> schrieb am So., 11. Okt. 2015
16:05:

> In src/javascript/app/ng-wig/ngWigEditableDirective.js
> <https://github.com/stevermeister/ngWig/pull/81#discussion_r41710786>:
>
> > @@ -14,7 +14,25 @@ angular.module('ngWig')
> >          ngModelController.$setViewValue($element.html());
> >        }
> >
> > -      $element.bind('blur keyup change paste focus click', function() {
> > +      var eventsToBind = [
> > +        'blur',
> > +        'keyup',
> > +        'change',
> > +        'focus',
> > +        'click'
> > +      ];
> > +
> > +      if (angular.isFunction(scope.onPaste)) {
> > +        $element.on('paste', function(e) {
> > +          scope.onPaste(e, $element.html()).then(function(val) {
>
> are you sure that it should be async operation?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/stevermeister/ngWig/pull/81/files#r41710786>.
>